src/sentry/mail/notifications.py

```diff
@@ -21,48 +21,48 @@ def get_headers(notification: BaseNotification) -> Mapping[str, Any]:
         "X-SMTPAPI": json.dumps({"category": notification.get_category()}),
     }
 
-    if notification.group:
+    group = getattr(notification, "group", None)
+    if group:
         headers.update(
             {
-                "X-Sentry-Logger": notification.group.logger,
-                "X-Sentry-Logger-Level": notification.group.get_level_display(),
-                "X-Sentry-Reply-To": group_id_to_email(notification.group.id),
+                "X-Sentry-Logger": group.logger,
+                "X-Sentry-Logger-Level": group.get_level_display(),
+                "X-Sentry-Reply-To": group_id_to_email(group.id),
             }
         )
 
     return headers
 
 
 def get_subject_with_prefix(
-    notification: BaseNotification, mail_option_key: Optional[str] = None
+    notification: BaseNotification,
+    context: Optional[Mapping[str, Any]] = None,
+    mail_option_key: Optional[str] = None,
 ) -> bytes:
     key = mail_option_key or "mail:subject_prefix"
     prefix = str(
         ProjectOption.objects.get_value(notification.project, key)
         or options.get("mail.subject-prefix")
     )
-    return f"{prefix}{notification.get_subject()}".encode("utf-8")
+    return f"{prefix}{notification.get_subject(context)}".encode("utf-8")
```

Comment:
was this (stuff about context) supposed to be in another PR?

Reply:
I think it makes sense to include here but I can extract it to the digest notification PR.